packages/flutter/test/material/range_slider_test.dart

```diff
@@ -2810,4 +2810,46 @@ void main() {
 
     semantics.dispose();
   }, semanticsEnabled: false);
+
+  // Regression test for hhttps://github.com/flutter/flutter/issues/161805
+  testWidgets('Discrete RangeSlider does not apply thumb padding in a non-rounded track shape', (
+    WidgetTester tester,
+  ) async {
+    const double sliderPadding = 24.0;
+    final ThemeData theme = ThemeData(
+      sliderTheme: const SliderThemeData(
+        // Thumb padding is applied based on the track height.
+        trackHeight: 100,
+        rangeTrackShape: RectangularRangeSliderTrackShape(),
+      ),
+    );
+
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: theme,
+        home: Material(
+          child: SizedBox(
+            width: 300,
+            child: RangeSlider(
+              values: const RangeValues(0, 100),
+              max: 100,
+              divisions: 100,
+              onChanged: (RangeValues value) {},
+            ),
+          ),
+        ),
+      ),
+    );
+
+    final MaterialInkController material = Material.of(tester.element(find.byType(RangeSlider)));
+
+    expect(
+      material,
+      paints
+        // Start thumb.
+        ..circle(x: sliderPadding, y: 300.0, color: theme.colorScheme.primary)
```

Comment:
Just a question, where do we get this 24?

Reply:
This comes from the track padding applied by base track shape in `RangeSlider` widget.

Here  `trackLeft` and `trackRight` are `24` pixels.
https://github.com/flutter/flutter/blob/234b50aa532b429bc9eb2e846d0a140850d6304f/packages/flutter/lib/src/material/slider_theme.dart#L2213-L2216

I made such track padding customizable in the `Slider` widget last year (https://github.com/flutter/flutter/pull/156143) but it's not yet implemented in `RangeSlider` but the I used the same wording here "slider padding" since it's effectively the same thing.  

I just filed an issue https://github.com/flutter/flutter/issues/165046and assigned. I will complete the `Slider` padding work for `RangeSlider`. Thanks for the reminder.

Reply:
Ah I see! Could you add a comment above the number? Thanks a lot for the explanation!

Reply:
Added a comment